app/src/main/java/com/github/sdpcoachme/database/FireDatabase.kt

```diff
@@ -46,69 +44,54 @@ class FireDatabase(databaseReference: DatabaseReference) : Database {
-            val hasCapacity = groupEvent.participants.size < groupEvent.maxParticipants
-            if (!hasCapacity) { // TODO for Bryan: this error prevention should be taken into the onclick of the register button
-                val failingFuture = CompletableFuture<Schedule>()
-                failingFuture.completeExceptionally(Exception("Group event is full"))
-                failingFuture
-            } else {
-                val updatedGroupEvent = groupEvent.copy(participants = groupEvent.participants + email)
-                setChild(groupEvents, groupEventId, updatedGroupEvent).thenCompose {
-                    getSchedule(email, EventOps.getStartMonday()).thenApply { s ->
-                        val updatedSchedule = s.copy(
-                            events = s.events + EventOps.groupEventsToEvents(listOf(updatedGroupEvent)),
-                            groupEvents = s.groupEvents + groupEventId
-                        )
-                        setChild(schedule, id, updatedSchedule)
-                        updatedSchedule
-                    }
-                }
+
+        return getSchedule(email).thenCompose { s ->
+            getGroupEvent(groupEventId).thenCompose { groupEvent ->
+                val updatedSchedule = s.copy(
+                    events = s.events + EventOps.groupEventsToEvents(listOf(groupEvent)),
+                    groupEvents = s.groupEvents + groupEventId
+                )
+                setChild(schedule, id, updatedSchedule).thenApply { updatedSchedule }
             }
         }
     }
 
-    override fun getSchedule(email: String, currentWeekMonday: LocalDate): CompletableFuture<Schedule> {
+    override fun addEventToSchedule(email: String, event: Event): CompletableFuture<Schedule> {
         val id = email.replace('.', ',')
-        return getChild(schedule, id).thenApply { it.getValue(Schedule::class.java)!! }
-            .exceptionally { Schedule() }
+        return getSchedule(email).thenCompose {
+            val updatedSchedule = it.copy(events = it.events + event)  // Add new events to the schedule
+            setChild(schedule, id, updatedSchedule).thenApply { updatedSchedule }// Update DB
+        }
     }
 
-    override fun getGroupEvent(groupEventId: String): CompletableFuture<GroupEvent> {
-        val id = groupEventId.replace('.', ',')
-        return getChild(groupEvents, id).thenApply { it.getValue(GroupEvent::class.java)!! }
-            .exceptionally { GroupEvent() }
+    override fun getSchedule(email: String): CompletableFuture<Schedule> {
+        val id = email.replace('.', ',')
+        return getChild(schedule, id).thenApply { it.getValue(Schedule::class.java)!! }
+            .exceptionally { Schedule() }
     }
 
     override fun getContactRowInfos(email: String): CompletableFuture<List<ContactRowInfo>> {
         return getUser(email).thenApply {
-            it.chatContacts.filterNotNull()
```

Comment:
Why was the `fllterNotNull()` removed ?

Reply:
chatContacts was of type List<String!>, no need to check for nullity here.